Cargo.toml

```diff
@@ -50,6 +50,7 @@ snafu = ">= 0.5, <= 0.6"
 getset = "0.0.9"
 enum-map = "0.6"
 triple_accel = "0.3.2"
+rayon = "1.4.0"
```

Comment:
rayon is a somewhat heavy dependency - I wonder if we should make a new feature called 'parallel'  or 'rayon' that let's the user opt-in to parallel implementation of algorithms, where available.